coreneuron/nrniv/nrnoptarg.cpp

```diff
@@ -29,306 +29,363 @@ THE POSSIBILITY OF SUCH DAMAGE.
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
+#include "coreneuron/nrnmpi/nrnmpi.h"
 #include "coreneuron/nrniv/nrnoptarg.h"
-#include "coreneuron/utils/sdprintf.h"
-
-extern "C" void nrn_exit(int);
-extern int nrnmpi_myid;
-
-cn_parameters::cn_parameters() {
-    tstart = 0.0;
-    tstop = 100.0;
-    dt = -1000.;
-
-    dt_io = 0.1;
-    dt_report = 0.1;
-
-    celsius = -1000.0;  // precedence: set by user, globals.dat, 34.0
-    voltage = -65.0;
-    maxdelay = 10.0;
-
-    forwardskip = 0;
-
-    spikebuf = 100000;
-    prcellgid = -1;
-
-    threading = 0;
-    compute_gpu = 0;
-    cell_interleave_permute = 0;
-    nwarp = 0; /* 0 means not specified */
-    report = 0;
-
-    patternstim = NULL;
-
-    datpath = ".";
-    outpath = ".";
-    filesdat = "files.dat";
-
-    multiple = 1;
-    extracon = 0;
-}
+#include "coreneuron/utils/ezoption/ezOptionParser.hpp"
+
+struct param_int {
+    const char* names; /* space separated (includes - or --) */
+    int dflt, low, high;
+    const char* usage;
+};
+
+struct param_dbl {
+    const char* names; /* space separated (includes - or --) */
+    double dflt, low, high;
+    const char* usage;
+};
+
+struct param_flag {
+    const char* names; /* space separated (includes - or --) */
+    const char* usage;
+};
+
+struct param_str {
+    const char* names; /* space separated (includes - or --) */
+    const char* dflt;
+    const char* usage;
+};
+
+static param_int param_int_args[] = {
+    {"--spikebuf -b", 100000, 1, 2000000000, "Spike buffer size. (100000)"},
+    {"--spkcompress", 0, 0, 100000, "Spike compression. Up to ARG are exchanged during MPI_Allgather. (0)"},
+    {"--prcellgid -g", -1, -1, 2000000000, "Output prcellstate information for the gid NUMBER."},
+    {"--cell-permute -R", 1, 0, 3,
+     "Cell permutation, 0 No; 1 optimise node adjacency; 2 optimize parent adjacency. (1)"},
+    {"--nwarp -W", 0, 0, 1000000, "Number of warps to balance. (0)"},
```

Comment:
0 (unused) is a good default. It takes some experimenting with balance to determine the optimal
value on gpu.